python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -0,0 +1,113 @@
+"""Process dosage matrix, GWAS summary statistics, and LD maps for PRS C+T calculation."""
+
+import glob
+import logging
+import re
+import pandas as pd
+from pyarrow import feather  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+def _load_preprocessed_dosage_matrix(dosage_matrix_path: str) -> pd.DataFrame:
+    """Load in dosage matrix generated from bystro-vcf."""
+    return feather.read_feather(dosage_matrix_path)
+
+
+def _load_association_scores(gwas_scores_path: str) -> pd.DataFrame:
+    """Load in GWAS summary statistics provided by user."""
+    return pd.read_csv(gwas_scores_path, delimiter="\t", comment="#")
+
+
+def _load_genetic_maps_from_feather(map_directory_path: str) -> dict[str, pd.DataFrame]:
+    """Load genetic maps from Feather files in the specified directory, using logging for messages.
+
+    Args:
+    ----
+    map_directory_path: The path to the directory containing Feather files.
+
+    Returns:
+    -------
+    A dictionary where keys are 'GeneticMap{chromosome_number}' and values are the corresponding
+    DataFrames loaded from Feather files.
+    """
+    file_pattern = f"{map_directory_path}/*.feather"
+    map_file_list = glob.glob(file_pattern)
+    genetic_maps = {}
+    for file in map_file_list:
+        try:
+            genetic_map = pd.read_feather(file)
+            match = re.search(r"chromosome_(\d+)_genetic_map", file)
+            if match:
+                chrom_num = match.group(1)
+                key = f"GeneticMap{chrom_num}"
+                genetic_maps[key] = genetic_map
+                logging.info("Successfully read %s from %s", key, file)
+            else:
+                logging.warning("Could not determine chromosome number from %s", file)
+        except Exception:
+            logging.exception("Failed to read from %s", file)
+    return genetic_maps
+
+
+def _preprocess_scores(gwas_scores_filepath: str) -> pd.DataFrame:
```

Comment:
If the goal of this is to generate a locus column in the GWAS covariates tables, this isn't the most optimal approach.  Instead, following the approaches described in https://stackoverflow.com/questions/39291499/how-to-concatenate-multiple-column-values-into-a-single-column-in-pandas-datafra, you should use Pandas vectorized functions:

df['CHR'] + ':' + df['POS'] + x["OTHER_ALLELE"] + x["EFFECT_ALLELE"]

after checking that those columns exist:

```py
columns = set(df.columns)
if "CHR" not in columns:
   raise ValueError('CHR does not exist')
etc

Reply:
```py
df['CHR'] = df['CHR'].as_type('str')
df['POS'] = df['POS'].as_type('str')
df['OTHER_ALLELE'] = df['OTHER_ALLELE'].as_type('str')
df['EFFECT_ALLELE'] = df['EFFECT_ALLELE'].as_type('str')

scores["ID_effect_as_alt"] = df['CHR'] + ':' + df['POS'] + x["OTHER_ALLELE"] + x["EFFECT_ALLELE"]
```